app/ante/eth.go

```diff
@@ -188,8 +191,8 @@ func (egcd EthGasConsumeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 
 		if ctx.IsCheckTx() {
 			// We can't trust the tx gas limit, because we'll refund the unused gas.
-			if txData.GetGas() > MaxTxGasWanted {
-				gasWanted += MaxTxGasWanted
+			if txData.GetGas() > egcd.maxGasWanted {
+				gasWanted += egcd.maxGasWanted
```

Comment:
can this cause non-determinism if nodes use different values?

Reply:
this is only for check tx, so I think it's ok.